priv/apps/itest/test/itest/in_flight_exits_test.exs

```diff
@@ -3,81 +3,925 @@ defmodule InFlightExitsTests do
+        keys: []
+      )
+
+    unsigned_txbytes = ExPlasma.Transaction.encode(unsigned_submitted_tx)
+
+    alice_state =
+      alice_state
+      |> Map.put(:submitted_tx, submitted_tx)
+      |> Map.put(:txbytes, txbytes)
+      |> Map.put(:unsigned_submitted_tx, unsigned_submitted_tx)
+      |> Map.put(:unsigned_txbytes, unsigned_txbytes)
+
+    entity = "Alice"
+    {:ok, Map.put(state, entity, alice_state)}
+  end
+
+  defthen ~r/Bob gets in flight exit data for "(?<amount>[^"]+)" ETH$/, %{amount: amount}, state do
+    amount = Currency.to_wei(amount)
+    %{address: bob_address, utxos: bob_utxos, pkey: bob_pkey} = bob_state = state["Bob"]
+
+    # inputs
+    bob_deposit_utxo = hd(bob_utxos)
+
+    bob_deposit_input = %ExPlasma.Utxo{
+      blknum: bob_deposit_utxo["blknum"],
+      currency: Currency.ether(),
+      oindex: 0,
+      txindex: 0,
+      output_type: 1,
+      owner: bob_address
+    }
+
+    # outputs
+    bob_output = %ExPlasma.Utxo{
+      currency: Currency.ether(),
+      owner: bob_address,
+      amount: amount
+    }
+
+    transaction = %Payment{inputs: [bob_deposit_input], outputs: [bob_output]}
+
+    submitted_tx =
+      ExPlasma.Transaction.sign(transaction,
+        keys: [bob_pkey]
+      )
+
+    txbytes = ExPlasma.Transaction.encode(submitted_tx)
+
+    unsigned_submitted_tx =
+      ExPlasma.Transaction.sign(transaction,
+        keys: []
+      )
+
+    unsigned_txbytes = ExPlasma.Transaction.encode(unsigned_submitted_tx)
+
+    payload = %InFlightExitTxBytesBodySchema{txbytes: Encoding.to_hex(txbytes)}
+    response = pull_api_until_successful(InFlightExit, :in_flight_exit_get_data, Watcher.new(), payload)
+    exit_data = IfeExitData.to_struct(response)
+
+    bob_state =
+      bob_state
+      |> Map.put(:exit_data, exit_data)
+      |> Map.put(:unsigned_txbytes, unsigned_txbytes)
+
+    entity = "Bob"
+    {:ok, Map.put(state, entity, bob_state)}
+  end
+
+  defthen ~r/Alice sends a transaction$/, _, state do
+    %{txbytes: txbytes} = alice_state = state["Alice"]
+
+    transaction_submit_body_schema = %TransactionSubmitBodySchema{transaction: Encoding.to_hex(txbytes)}
+    {:ok, response} = Transaction.submit(Watcher.new(), transaction_submit_body_schema)
+
+    submit_transaction_response =
+      response
+      |> Map.get(:body)
+      |> Jason.decode!()
+      |> Map.get("data")
+      |> SubmitTransactionResponse.to_struct()
+
+    alice_state = Map.put(alice_state, :transaction_submit, submit_transaction_response)
+
+    entity = "Alice"
+
+    {:ok, Map.put(state, entity, alice_state)}
+  end
+
+  defwhen ~r/Bob sends a transaction spending Alices output$/, _, state do
+    %{address: alice_address, transaction_submit: alice_transaction_submit} = state["Alice"]
+
+    %{address: bob_address, pkey: bob_pkey} = bob_state = state["Bob"]
+
+    # inputs
+    bob_input = %ExPlasma.Utxo{
+      blknum: alice_transaction_submit.blknum,
+      currency: Currency.ether(),
+      oindex: 1,
```

Comment:
hm... yes probably! this was picked from here: https://github.com/omisego/elixir-omg/pull/1252/files#diff-efa6ec8bc1db68317994ddee8f513d1cL67

Reply:
ah right, yeah, that was a bit clunky. Now given the increasing expressiveness of the tool, I'd switch to using `_submit.oindex` etc right away

Reply:
`oindex` is not present in the SubmitTransactionResponse API `/transaction.submit`, but `txindex` is.

Reply:
```
%Itest.ApiModel.SubmitTransactionResponse{
  blknum: 1000,
  txhash: "0xc52acc43d6280d40245e99c20f6985167b1bb61eb9e98343015b72f9986f967d",
  txindex: 0
}
```

Reply:
ah, of course :man_facepalming:, my bad, this is a per-transaction response. So you get to know which `oindex` you want during creation of the transaction, not from the response.

I guess, since this is not a simple swap as I wrongly thought, it can be as well left for later refactors.